pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1267,19 +1270,34 @@ class Timedelta(_Timedelta):
                                  "(days,seconds....)")
 
             kwargs = {key: _to_py_int_float(kwargs[key]) for key in kwargs}
-
-            nano = convert_to_timedelta64(kwargs.pop('nanoseconds', 0), 'ns')
-            try:
-                value = nano + convert_to_timedelta64(timedelta(**kwargs),
-                                                      'ns')
-            except TypeError as e:
+            if not cls._req_any_kwargs_new.intersection(kwargs):
                 raise ValueError(
                     "cannot construct a Timedelta from the passed arguments, "
                     "allowed keywords are "
                     "[weeks, days, hours, minutes, seconds, "
                     "milliseconds, microseconds, nanoseconds]"
                 )
 
+            # GH43764, making sure any nanoseconds contributions from any kwarg
+            # is taken into consideration
+            seconds = int((
+                (
+                    (kwargs.get('days', 0) + kwargs.get('weeks', 0) * 7) * 24
+                    + kwargs.get('hours', 0)
+                ) * 3600
+                + kwargs.get('minutes', 0) * 60
+                + kwargs.get('seconds', 0)
+                ) * 1_000_000_000
+            )
+
+            value = convert_to_timedelta64(
```

Comment:
let's just directly create a `np.timedelta64(ts, "ns")` no reason to go thru the routine when we know exactly what we have.

Reply:
Nice. Oversaw that. Performance is even better now.